backend/app/services/chat/url_parser.py

```
@@ -0,0 +1,740 @@
+                url=url,
+                type=UrlType.IMAGE,
+                content=data_url,
+                size=content_length,
+            )
+
+        except Exception as e:
+            logger.exception(f"Error parsing image {url}")
+            return ParsedUrlResult(
+                url=url,
+                type=UrlType.IMAGE,
+                error=f"Failed to parse image: {str(e)}",
```

Ruff (0.14.10)

Use explicit conversion flag

Replace with conversion flag

(RUF010)